src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/wa/WaTaskCreationCronForExpiredBfActions.java

```diff
@@ -0,0 +1,136 @@
+
+    @Value("${cron.maxCasesPerSearch}")
+    private int maxCases;
+
+    private List<Long> processedCaseIdsToSkip = new ArrayList<>();
+
+    @Override
+    public void run() {
+        log.info("WaTaskCreationCronForExpiredBfActions is running");
+        if (!featureToggleService.isWorkAllocationEnabled()
+                || !featureToggleService.isWaTaskForExpiredBfActionsEnabled()) {
+            log.info("WaTaskCreationCronForExpiredBfActions is disabled by feature toggle");
+            return;
+        }
+
+        log.info("In WaTaskCreation ... cron job - Checking for expired BFDates");
+        String[] caseTypeIds = caseTypeIdsString.split(",");
+        String adminUserToken = adminUserService.getAdminUserToken();
+        String today = UtilHelper.formatCurrentDate2(LocalDate.now());
+        String query = buildQueryForExpiredBFActions(getEffectiveYesterday(), today);
+
+        Arrays.stream(caseTypeIds).forEach(caseTypeId -> {
+            try {
+                List<SubmitEvent> cases = ccdClient.buildAndGetElasticSearchRequest(adminUserToken, caseTypeId, query);
+                while (CollectionUtils.isNotEmpty(cases)) {
+                    cases.stream()
+                            .filter(o ->
+                                    !processedCaseIdsToSkip.contains(o.getCaseId()))
+                            .forEach(o -> {
+                                triggerTaskEventForCase(adminUserToken, o, caseTypeId);
+                                log.info("Triggered WA task for case ID: {} in case type: {}",
+                                        o.getCaseId(), caseTypeId);
+                                processedCaseIdsToSkip.add(o.getCaseId());
+                            });
+                    cases = ccdClient.buildAndGetElasticSearchRequest(adminUserToken, caseTypeId, query);
+                }
+            } catch (Exception e) {
+                log.error(e.getMessage());
+            }
+        });
+        log.info("WaTaskCreationCronForExpiredBfActions processed {} case IDs", processedCaseIdsToSkip.size());
+        log.info("WaTaskCreationCronForExpiredBfActions completed execution");
+    }
+
+    private String getEffectiveYesterday() {
+        // Determine the effective "yesterday" based on the current day of the week
+        LocalDate today = LocalDate.now();
+        DayOfWeek dayOfWeek = today.getDayOfWeek();
+        LocalDate effectiveYesterday;
+
+        switch (dayOfWeek) {
+            case DayOfWeek.MONDAY -> {
+                // If today is Monday, go back to Friday
+                effectiveYesterday = today.minusDays(3);
+            }
+            case DayOfWeek.SUNDAY -> {
+                // If today is Sunday, go back to Friday as well
+                effectiveYesterday = today.minusDays(2);
+            }
+            default -> {
+                // Regular yesterday
+                effectiveYesterday = today.minusDays(1);
+            }
+        }
```

Comment:
Just tidying the code up
```suggestion
        LocalDate effectiveYesterday = switch (dayOfWeek) {
            case MONDAY -> today.minusDays(3); // If today is Monday, go back to Friday
            case SUNDAY -> today.minusDays(2); // If today is Sunday, go back to Friday as well
            default -> today.minusDays(1);     // Regular yesterday
        };```